invenio_rdm_migrator/streams/actions/drafts.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM migration drafts row load module."""
+
+from datetime import datetime
+from functools import partial
+
+from ...actions import LoadAction
+from ...load.ids import generate_pk, generate_recid, generate_uuid
+from ...load.postgresql.transactions.operations import Operation, OperationType
+from ..models.files import FilesBucket
+from ..models.pids import PersistentIdentifier
+from ..models.records import RDMDraftMetadata, RDMVersionState
+from ..records.table_generators.parents import generate_parent_rows  # FIXME: move
+from ..records.table_generators.references import (
+    CommunitiesReferencesMixin,
+    PIDsReferencesMixin,
+)
+
+
+class RDMDraftCreateAction(LoadAction, CommunitiesReferencesMixin, PIDsReferencesMixin):
+    """RDM draft creation."""
+
+    name = "create-draft"
+
+    def __init__(
+        self,
+        tx_id,
+        pid,
+        bucket,
+        draft,
+        parent,
```

Comment:
minor: I'm a bit skeptical on having multiple data variable parameters mixed with the state params in the class constructor... Could we address this by introducing a "target data class" defined next to the `LoadAction` and use it for wrapping the data returned for the `TransformAction`. I was thinking about something like:

```python
# invenio_rdm_migrator/streams/actions/drafts.py

class RDMDraftCreateAction(...):
    ...

@dataclass
class RDMDraftCreateData
    tx_id: int
    pid: dict
    bucket: dict
    draft: dict
    parent: dict

    _action_cls = RDMDraftCreateAction  # here's how we know the mapping 

# invenio_rdm_migrator/actions/base.py
class DraftCreateAction(TransformAction):

    mapped_cls = RDMDraftCreateData

# zenodo_rdm_migrator/streams/actions/trasnform/drafts.py
class DraftCreateAction(TransformAction):

    mapped_cls = RDMDraftCreateData
```

An alternative, which we also did for the snapshot loading, was that we just passed a big `dict` with everything inside (but in that case, we had specific streams that knew the entire pipeline in terms of what to expect in this `dict` on the Load step).

Reply:
_Pending implementation_

Reply:
Done by @slint